hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/fsck/ContainerMapper.java

```diff
@@ -79,6 +80,22 @@ public static void main(String[] args) throws IOException {
     }
     OmMetadataManagerImpl metadataManager =
         new OmMetadataManagerImpl(configuration, null);
+    return getBlockIdDetailsUtils(metadataManager);
+  }
+
+  public Map<Long, List<Map<Long, BlockIdDetails>>>
+      parseOmDB(OzoneConfiguration configuration, OMPerformanceMetrics perfMetrics) throws IOException {
+    String path = configuration.get(OZONE_OM_DB_DIRS);
+    if (path == null || path.isEmpty()) {
+      throw new IOException(OZONE_OM_DB_DIRS + "should be set ");
+    }
+    OmMetadataManagerImpl metadataManager =
+        new OmMetadataManagerImpl(configuration, null, perfMetrics);
+    return getBlockIdDetailsUtils(metadataManager);
+  }
+
+  private static Map<Long, List<Map<Long, BlockIdDetails>>> getBlockIdDetailsUtils(
+      OmMetadataManagerImpl metadataManager) throws IOException {
```

Comment:
This will be unnecessary if `OmMetadataManagerImpl` is fixed to handle `null` metrics.

```suggestion
```

Reply:
Noted, removed the repetitive code by changing approach.

Reply:
@adoroszlai, please have a look at update code block.